src/extension/feature-pregeneralized/src/main/java/org/geoserver/data/gen/info/GeneralizationInfosProviderImpl.java

```diff
@@ -35,10 +45,10 @@ protected URL deriveURLFromSourceObject(Object source) throws IOException {
 
             GeoServerResourceLoader loader =
                     GeoServerExtensions.bean(GeoServerResourceLoader.class);
-            File f =
-                    Resources.find(
-                            Resources.fromURL(Files.asResource(loader.getBaseDirectory()), path),
-                            true);
+            Resource resource = loader.get(Paths.convert(path));
+
+            File f = Resources.find(resource);
+
```

Comment:
```suggestion
            URL url = null;
            if( resource.getType() == ResourceType.FILE){
                url = URLs.fileToUrl(resource.file());
            }
            else {
               url = new URL(path);
            }
```

Reply:
There is no FILE value for the enum Type of Resource and implementing it requires quite a work

Reply:
Type.RESOURCE is a better match

Reply:
I have take your comments into account as mus as possible.

Reply:
I didn't spot the empty file : it is removed now.